### ISSUE TITLE
kernel BUG at net/core/skbuff.c:2437

### ISSUE THREAD

**Issue by keshonok** [natsys/tempesta#435]
This happens on the latest Tempesta kernel with the SKB page allocator, and Tempesta version that has `unref_data` always set to `false`, even in the case where usually set to `true`. The test was over a 10G link.

The trace shows that the 

```
[52518.672355] kernel BUG at net/core/skbuff.c:2437!
[52518.677478] invalid opcode: 0000 [#1] SMP 
[52518.682041] Modules linked in: tfw_sched_rr(O) tfw_sched_http(O) tfw_sched_hash(O) tempesta_fw(O) tempesta_db(O) nls_utf8 fuse btrfs xor raid6_pq ntfs vfat msdos fat dm_mod nfsd auth_rpcgss oid_registry nfs_acl nfs lockd grace fscache sunrpc snd_pcm ast ttm intel_powerclamp coretemp snd_timer crct10dif_pclmul snd crc32_pclmul drm_kms_helper aesni_intel drm aes_x86_64 lrw gf128mul i2c_algo_bit glue_helper iTCO_wdt joydev psmouse soundcore ablk_helper iTCO_vendor_support cryptd i2c_i801 lpc_ich evdev i2c_core syscopyarea mfd_core serio_raw sysfillrect tpm_tis sysimgblt tpm pcspkr button acpi_cpufreq processor i7core_edac shpchp edac_core thermal_sys ipmi_watchdog ipmi_si ipmi_poweroff ipmi_devintf ipmi_msghandler autofs4 ext4 crc16 mbcache jbd2 sg sr_mod sd_mod cdrom hid_generic usbhid hid ehci_pci uhci_hcd ahci libahci ehci_hcd e1000e ixgbe libata crc32c_intel mdio dca usbcore ptp scsi_mod usb_common pps_core [last unloaded: tempesta_db]
[52518.771363] CPU: 1 PID: 0 Comm: swapper/1 Tainted: G        W  O    4.1.12+ #1
[52518.779233] Hardware name: Quanta                 S89/QR1-890/QSSC-890  /S89/QR1-890/QSSC-890  , BIOS S89_3A10 10/21/2010
[52518.790870] task: ffff8800590fa290 ti: ffff8800590fc000 task.ti: ffff8800590fc000
[52518.799054] RIP: 0010:[<ffffffff814dc650>]  [<ffffffff814dc650>] skb_copy_and_csum_bits+0x2e0/0x2f0
[52518.808835] RSP: 0018:ffff88007a003a58  EFLAGS: 00010202
[52518.814873] RAX: ffff880018bccac0 RBX: 000000000000017a RCX: 00000000000002c0
[52518.822745] RDX: 0000000000000002 RSI: ffff880018bcc800 RDI: ffff880018bccac0
[52518.830624] RBP: ffff88007a003aa8 R08: 65683c0a3e6c6d74 R09: ffff880018bcc6f8
[52518.838504] R10: 00000000000001ae R11: 7469743c0a3e6461 R12: 0000000000000076
[52518.846413] R13: 00000000000001ae R14: 0000000000000000 R15: 00000000000001ae
[52518.854313] FS:  0000000000000000(0000) GS:ffff88007a000000(0000) knlGS:0000000000000000
[52518.863177] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[52518.869702] CR2: 00007f3dfbe5a000 CR3: 0000000001c0c000 CR4: 00000000000006e0
[52518.877632] Stack:
[52518.880431]  ffff880018bcc6f8 ffff8800000001ae ffff88004ab026da e115dfa3000001ae
[52518.888705]  ffffffff8153e6d1 ffff880058e52600 ffff88004ab02400 0000000000000000
[52518.896981]  000000000000022c ffff880057da5d28 ffff88007a003ad8 ffffffff81577e61
[52518.905253] Call Trace:
[52518.908506]  <IRQ> 
[52518.910445]  [<ffffffff8153e6d1>] ? ip_setup_cork+0x141/0x150
[52518.918024]  [<ffffffff81577e61>] icmp_glue_bits+0x31/0x90
[52518.924345]  [<ffffffff81540ea9>] __ip_append_data.isra.44+0x849/0xae0
[52518.931724]  [<ffffffff81577e30>] ? icmp_push_reply+0x120/0x120
[52518.938506]  [<ffffffff81577e30>] ? icmp_push_reply+0x120/0x120
[52518.945290]  [<ffffffff815411ba>] ip_append_data.part.45+0x7a/0xf0
[52518.952326]  [<ffffffff815426b4>] ip_append_data+0x34/0x40
[52518.958655]  [<ffffffff81577d7a>] icmp_push_reply+0x6a/0x120
[52518.965149]  [<ffffffff81579729>] icmp_send+0x5a9/0x610
[52518.971220]  [<ffffffff815338d2>] ipv4_link_failure+0x22/0xa0
[52518.977811]  [<ffffffff81575e2a>] arp_error_report+0x3a/0x80
[52518.984324]  [<ffffffff814ff8a1>] neigh_invalidate+0x91/0x110
[52518.990930]  [<ffffffff81501aa4>] neigh_timer_handler+0x1f4/0x2a0
[52518.997897]  [<ffffffff815018b0>] ? neigh_seq_start+0x1f0/0x1f0
[52519.004699]  [<ffffffff810ee354>] call_timer_fn+0xb4/0x4b0
[52519.011071]  [<ffffffff810ee2a5>] ? call_timer_fn+0x5/0x4b0
[52519.017525]  [<ffffffff815018b0>] ? neigh_seq_start+0x1f0/0x1f0
[52519.024324]  [<ffffffff810ee98c>] run_timer_softirq+0x23c/0x480
[52519.031112]  [<ffffffff8106d2a4>] __do_softirq+0xf4/0x630
[52519.037372]  [<ffffffff8106d985>] irq_exit+0xa5/0xb0
[52519.043197]  [<ffffffff81629096>] smp_apic_timer_interrupt+0x46/0x60
[52519.050400]  [<ffffffff816274a0>] apic_timer_interrupt+0x70/0x80
[52519.057226]  <EOI> 
[52519.059164]  [<ffffffff814b55bd>] ? cpuidle_enter_state+0xad/0x360
[52519.067106]  [<ffffffff814b55b8>] ? cpuidle_enter_state+0xa8/0x360
[52519.074041]  [<ffffffff814b58a7>] cpuidle_enter+0x17/0x20
[52519.080175]  [<ffffffff810b8273>] cpu_startup_entry+0x473/0x5a0
[52519.086806]  [<ffffffff8103aaf5>] start_secondary+0x185/0x1b0
[52519.093234] Code: 09 00 00 48 c7 c7 ec 59 a4 81 44 89 55 b8 e8 48 ab b8 ff 44 8b 55 b8 e9 4f ff ff ff 0f 1f 80 00 00 00 00 45 31 ff e9 a8 fd ff ff <0f> 0b 45 89 f2 e9 25 ff ff ff 66 0f 1f 44 00 00 66 66 66 66 90 
[52519.114571] RIP  [<ffffffff814dc650>] skb_copy_and_csum_bits+0x2e0/0x2f0
[52519.121979]  RSP <ffff88007a003a58>
```

**Comment by keshonok** [comment]
This particular bug occurs here: [ss_skb.c:506](/natsys/tempesta/tree/c86b30862210b3ea988fa1b93db3428695eeafb0/tempesta_fw/ss_skb.c#L506). Also, it appears that the use of `skb_dst` in another place in this function is questionable as well.

**Comment by keshonok** [comment]
Fixed with 95e6b79.
